quixstreams/models/serializers/protobuf.py

```diff
@@ -0,0 +1,100 @@
+from typing import Union, Mapping, Iterable, Dict
+
+from .base import Serializer, Deserializer, SerializationContext
+from .exceptions import SerializationError
+
+from google.protobuf.message import Message, DecodeError, EncodeError
+from google.protobuf.json_format import MessageToDict, ParseDict, ParseError
+
+__all__ = ("ProtobufSerializer", "ProtobufDeserializer")
+
+
+class ProtobufSerializer(Serializer):
+    def __init__(
+        self,
+        msg_type: Message,
+        deterministic: bool = False,
+        ignore_unknown_fields: bool = False,
+    ):
+        """
+        Serializer that returns data in protobuf format.
+
+        Serialisation from a python dictionary can have a significant performance impact. An alternative is to pass the serializer an object of the `msg_type` class.
+
+        :param msg_type: protobuf message class.
+        :param deterministic: If true, requests deterministic serialization of the protobuf, with predictable ordering of map keys
+            Default - `False`
+        :param ignore_unknown_fields: If True, do not raise errors for unknown fields.
+            Default - `False`
+        """
+        super().__init__()
+        self._msg_type = msg_type
+
+        self._deterministic = deterministic
+        self._ignore_unknown_fields = ignore_unknown_fields
+
+    def __call__(
+        self, value: Union[Dict, Message], ctx: SerializationContext
+    ) -> Union[str, bytes]:
+
+        try:
+            if isinstance(value, self._msg_type):
+                return value.SerializeToString(deterministic=self._deterministic)
+
+            msg = self._msg_type()
+            return ParseDict(
+                value, msg, ignore_unknown_fields=self._ignore_unknown_fields
+            ).SerializeToString(deterministic=self._deterministic)
+        except (EncodeError, ParseError) as exc:
+            raise SerializationError(str(exc)) from exc
+
+
+class ProtobufDeserializer(Deserializer):
+    def __init__(
+        self,
+        msg_type: Message,
+        use_integers_for_enums: bool = False,
+        preserving_proto_field_name: bool = False,
+        to_dict=True,
```

Comment:
The type hint is missing

Reply:
The docs also must mention that in the current implementation, `to_dict` affects speed.